### PR TITLE
Sync site/region/world structures to Toady headers

### PR DESCRIPTION
**Pull request by quietust**
Almost done...

### CODE REVIEW

**Comment by quietust** [comment]
Note: double-check battlefield, location_scribe_jobs, site_realization_building, and world_site_realization to confirm that everything looks properly aligned

**Review by myk002 — APPROVED** [review]
this is huge! awesome work!

I'll merge once I finish adapting the codebase and verify the structures in-game

**Comment by myk002** [comment]
location_scribe_jobs looks good

something's up with battlefield; I'll double check the headers

forts don't seem to have world site realizations. I'll check that against an adventure mode site in a bit

**Comment by myk002** [comment]
ok, it looks like I've already done site_realization_building and world_site_realization in the adv-beta branch. I'll reconcile those

File: df.world.xml [hunk 1]
```diff
@@ -993,9 +993,7 @@
         <compound name='enemy_status_cache'> bay12: unit_reaction_handlerst
             <static-array name='slot_used' type-name='bool' count='500'/>
             <static-array name='rel_map' count='500'>
-                <static-array count='500'> bay12: unit_reactionst
-                    <enum name='reaction' type-name='unit_reaction_type'/>
-                </static-array>
+                <static-array type-name='unit_reaction_type' count='500'/> bay12: unit_reactionst
```
Comment (quietust) [inline df.world.xml:996]: This one was actually intentional - it's not a 500x500 array of enums, but a 500x500 array of **structures** which happen to contain nothing but an enum.

Comment (myk002) [reply df.world.xml:996]: the compiler didn't seem to like it. it couldn't initialize/address it properly

**Pull request merged**